File: C/CS50/Week_2/readability/readability.c
#include <stdio.h>
#include <string.h> // I can use the strlen function which counts characters

int main(void)
{
// Defines variables
    char string[999];
    int characters;
    int spaces = 0;
    int letters = 0;
    int sentences = 0;
    float score;
    float l = 0;
    float s = 0;

// Takes user input
    printf("Write some letters:");
    fgets(string, sizeof(string), stdin); // Can recieve input with spaces - However, can only be used on string

// Counts total characters
    characters = strlen(string);

// Count total spaces
    for (int i = 0; string[i] != '\0'; i++) // Itterate trough until \0
    {
        if (string[i] == ' ')
        {
            spaces++; // Spaces are also a measure for total words (+1 for the first word)
        }
    }

    // Counts sentences
    for (int j = 0; string[j] != '\0'; j++)
    {
        if (string[j] == ('.') || string[j] == ('!') || string[j] == ('?'))
        {
            sentences++;
        }
    }

    // Counts all letters
    for (int t = 0; string[t] != '\0'; t++)
    {
        if ((string[t] >= 'a' && string[t] <= 'z') || (string[t] >= 'A' && string[t] <= 'Z'))
        {
            letters++;
        }
    }

    // Calculates readability
    // Average number of letters per 100 words
    l = (((float)letters / ((float)spaces + 1)) * 100);

    // Average number of sentences per 100 words
    s = (((float)sentences / ((float)spaces + 1)) * 100);

    // Calculation
    score = (0.0588 * (float)l) - (0.296 * (float)s) - 15.8;

// Gives output
    if (score > 2 && score < 17)
    {
        printf("Grade %0.f\n", score);
    }
    else if (score < 2)
    {
        printf("Before grade 1\n");
    }
    else
    {
        printf("Grade 16+\n");
    }
}

File: Course - CS50 (C, javascript, SQL, HTML, Python)/Week_6/pset/readability.py
from cs50 import get_string
import re


## Input
def main():
    while True:
        text = get_string("Write some letters:\n")
        break    

    ## Letters
    letters = len(re.sub("[^a-zA-Z]", "", text))

    words = len(text.split())

    sentences = text.count(".") + text.count("!") + text.count("?")

    grade = calculate(letters, words, sentences)

    if grade > 2 or grade < 16:
        print(f"Grade {round(grade)}")
    elif grade < 2:
        print("Before grade 1")
    else:
        print("Grade 16+")
    print(f"Number of letters: {letters}.\nNumber of words: {words}.\nNumber of sentences:{sentences} ")



def calculate(letters, words, sentences):
    l = ((letters / (words)) * 100)
    s = ((sentences / (words)) * 100)
    print(f"l is {l}, while s is {s}")
    return((0.0588 * l) - (0.296 * s) - 15.8)


main()